lib/cchain/provider/abci.go

```diff
@@ -763,3 +671,134 @@ func attsFromAtHeight(ctx context.Context, cl atypes.QueryClient, chainVer xchai
 
 	return atts, true, nil
 }
+
+// binarySearch uses a binary search between defined start and end consensus block heights to
+// find the attestations with the provided fromOffset.
+func binarySearch(
```

Comment:
binary search was extracted from search history function but it is unchanged (beside steps count part of for loop and metric report)

Reply:
in function comments, focus on what it returns, from your comment it looks like it returns attestations?